src/main/java/io/pravega/perf/PravegaBatchReaderWorker.java

```diff
@@ -0,0 +1,66 @@
+package io.pravega.perf;
+
+import io.pravega.client.BatchClientFactory;
+import io.pravega.client.batch.SegmentIterator;
+import io.pravega.client.batch.SegmentRange;
+import io.pravega.client.stream.impl.ByteArraySerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Iterator;
+import java.util.List;
+
+public class PravegaBatchReaderWorker extends ReaderWorker {
+    private static Logger log = LoggerFactory.getLogger(PravegaBatchReaderWorker.class);
+
+    private final BatchClientFactory batchClientFactory;
+    private final Iterator<SegmentRange> assignedSegments;
+
+    private SegmentIterator<byte[]> currentSegmentIterator;
+    private SegmentRange currentRange;
+    private boolean finished;
+
+    PravegaBatchReaderWorker(int readerId, int events, int secondsToRun, long start, PerfStats stats, String readerGrp, int timeout, boolean writeAndRead, BatchClientFactory batchClientFactory, List<SegmentRange> assignedSegments) {
+        super(readerId, events, secondsToRun, start, stats, readerGrp, timeout, writeAndRead);
+        this.batchClientFactory = batchClientFactory;
+
+        this.assignedSegments = assignedSegments.iterator();
+    }
+
+    @Override
+    public byte[] readData() {
+        if (finished) {
+            return null;
```

Comment:
This will not cause the benchmark process to stop. When all data has been read, the benchmark process will continue to run, reporting 0 events/sec. Is this desired? Maybe you want to throw an exception when the worker completes reading?

Reply:
Resolved.  BatchReaders now throw a `WorkerCompletedException` when all assigned events have been completed which stops the test early if there are no more events.